torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -1,191 +1,21 @@
 import abc
 import csv
 import importlib
-import itertools
-import os
 import pathlib
-from typing import Any, Dict, List, Optional, Sequence, Union, Tuple, Collection, Iterator
+from typing import Any, Dict, List, Optional, Sequence, Union, Collection, Iterator
 
 from torch.utils.data import IterDataPipe
-from torchvision._utils import sequence_to_str
 from torchvision.datasets.utils import verify_str_arg
-from torchvision.prototype.utils._internal import FrozenBunch, make_repr, add_suggestion
 
-from .._home import use_sharded_dataset
-from ._internal import BUILTIN_DIR, _make_sharded_datapipe
 from ._resource import OnlineResource
 
 
-class DatasetConfig(FrozenBunch):
-    # This needs to be Frozen because we often pass configs as partial(func, config=config)
-    # and partial() requires the parameters to be hashable.
-    pass
+def read_categories_file(path: pathlib.Path) -> List[List[str]]:
```

Comment:
Maybe we can move this into `_internal` and provide the extra convenience to load the file by only supplying the dataset name? I don't think we need to expose this, given that without sugar, the value of this function is quite minimal.